### DISCUSSION TITLE
Alist 探讨

### DISCUSSION THREAD

**Discussion by MiKoto-Railgun** [dyphire/mpv-config#145] [General]
大家有用alist的吗？一般有用mpv看alist的不，大家怎么看的，我现在用的是raidriver添加到本地然后用文件管理器查看（这样有播放列表，或者粘贴网址也行也有列表），又或者”使用MPV播放“脚本调用命令行打开（无播放 列表）

**Comment by sf467** [comment]
电脑的话，可以先在 alist 的界面里，复制视频文件的地址，然后在终端里输入
mpv "视频地址“
就可以直接用 mpv 播放了
当然有些时候会有限速的问题，那就只能先用多线程下载工具把视频下载下来，然后才能用 mpv 播放
不过听说 alist 内部有一个多线程缓存视频的功能，可以缓解限速的问题，但是不知道怎么启用

**Comment by dyphire** [comment]
可以尝试使用 https://github.com/VirtualHotBar/NetMount ，据说比 raidriver 好

**Comment by sf467** [comment]
我推荐一个 rclone 吧，这个主要是命令行的，可以进行云盘到本地的挂载，支持 win,linux,mac
性能不错，可以通过虚拟文件系统，将对云盘挂载到本地的文件的读写操作缓存到本地进行，更重要的是，这个缓存可以仅仅是”只缓存到对这个文件读取到的位置“，而不是”一旦有读写操作就缓存整个文件“

**Comment by wathper** [comment]
我使用 RaiDrive 将 alist 挂载到本地，使用 OneCommander管理器 来浏览，因为 webDAV 的网络波动大，一但出现严重延迟，系统文件管理器容易在加载目录的过程中卡死。

**Comment by aprils148** [comment]
之前用過alist配合[mpv-file-browser](https://github.com/CogentRedTester/mpv-file-browser)的ftp功能,參考看看

**Reply by HedioKojima** [reply]
https://github.com/CogentRedTester/mpv-file-browser  如果直接支持webdav那就省事多了

**Comment by HedioKojima** [comment]
使用gemini pro生成了一个webdav.lua，竟然可以用，Gemini的代码也是越来越好了https://gist.github.com/HedioKojima/fdbfdd73570650b01c809afb5ae7829b

**Reply by moxun33** [reply]
请问这个如何使用？

**Reply by HedioKojima** [reply]
将lua文件放入你本地的scripts文件夹，lua文件的local opts处修改为自己的webdav地址，用户名，密码。 打开mpv，摁 默认快捷键w，就可以了。

**Reply by Ladersbt** [reply]
> 将lua文件放入你本地的scripts文件夹，lua文件的local opts处修改为自己的webdav地址，用户名，密码。 打开mpv，随意播放一个视频或图片（纯ai生成的有点bug），摁 默认快捷键w，就可以了。

或许可以在脚本里加两行，以支持在配置文件里修改，而且这个脚本似乎在mpv空闲状态下使用会失败报错

**Reply by HedioKojima** [reply]
是的，在空闲状态下会失败。必须播放点东西。我对代码是一窍不通，如果您能够进行修改那就更好了。

**Reply by Ladersbt** [reply]
> 是的，在空闲状态下会失败。必须播放点东西。我对代码是一窍不通，如果您能够进行修改那就更好了。

我对代码基本也是不太懂，很多问题都是问的ai，空闲状态播放失败这个是可以解决的，比如你发布的 'webdav.lua osd by Gemini pro'，在103行加个 'playback_only = false'  ，即改为 'local res = mp.command_native({ name = "subprocess", playback_only = false, capture_stdout = true, args = args })' ，应该就可以了

**Reply by Ladersbt** [reply]
哦不对，应该不止这一处，因为你还写了个删除的功能，所以那里类似的也要加一个playback_only = false

**Reply by HedioKojima** [reply]
gist处的脚本已更改，空闲状态也可以使用了，感谢您的修改。还有那些需要修改的您也可以继续指正。🙏

**Reply by MiKoto-Railgun** [reply]
报个bug

<div style="display: flex; justify-content: center; gap: 10px;">
  <figure style="text-align: center; margin: 0;">
    <img src="https://github.com/user-attachments/assets/ba95d6dc-817b-428b-82ae-3b5091f89189" width="600" alt="图1">
  </figure>
  <figure style="text-align: center; margin: 0;">
    <img src="https://github.com/user-attachments/assets/f3066a1c-24da-42a2-b6b9-3aba0e9ce6a2" width="600" alt="图2">
  </figure>
</div>

命名里含有“&”的目录并不显示文件，可能与转义字符解析有关

<img width="960" height="540" alt="image" src="https://github.com/user-attachments/assets/5b60b1d1-7210-45a2-91bd-c01b30b8ecbf" />

**Reply by MiKoto-Railgun** [reply]
除此之外，确实不错

**Reply by HedioKojima** [reply]
嗯，好的。uosc版本的跑的ai版本比较早，我之后再让ai跑一下。还有uosc版本的删除文件的逻辑有点麻烦，还得改进。感谢您的指正

**Reply by HedioKojima** [reply]
uosc版已修正，我测试了一下，可以了

**Reply by HedioKojima** [reply]
webdav.lua uosc版 推荐使用此处的 https://github.com/Ladersbt/mpv-config/blob/master/scripts/uosc_webdav.lua
